common/static/js/src/ReactRenderer.jsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+class ReactRendererException extends Error {
+  constructor (...args) {
+    super(...args);
+    this.name = 'ReactRendererException';
+    Error.captureStackTrace(this, ReactRendererException);
+  }
+  toString() {
+    return `${this.name}: ${this.message}`;
+  }
+}
+
+export class ReactRenderer {
+  constructor({component, selector, componentName, props}) {
+    Object.assign(this, {
+      component,
+      selector,
+      componentName,
+      props,
+    });
+    this.handleArgumentErrors();
+    this.targetElement = this.getTargetElement(selector);
+    this.renderComponent();
+  }
+
+  handleArgumentErrors() {
+    if (this.component === null) {
+      throw new ReactRendererException(
+        `Component ${this.componentName} is not defined. Make sure you're ` +
+        `using a non-default export statement for the ${this.componentName} ` +
+        `class, that ${this.componentName} has an entry point defined ` +
+        `within the 'entry' section of webpack.common.config.js, and that the ` +
+        `entry point is pointing at the correct file path.`
+      );
+    }
+    if (this.props && !((!!this.props) && (this.props.constructor === Object))) {
```

Comment:
@ormsbee @jaebradley yeah, there's no explanation behind the duplicate falsy checks besides "I wasn't paying close enough attention." I want to throw if:
* something was passed to props
* that something is not an Object (functions and arrays don't count)

I think Jae's solution works best here since it catches the `null` case, so I'm going to go with that.